src/MSBuild.Abstractions/MSBuildHelpers.cs

```diff
@@ -374,6 +411,13 @@ private static bool UnquoteString(ref string s)
             }
 
             s = s[1..^1];
+
+            // Make sure there wasn't another string in there
+            if (s.Contains('\''))
```

Comment:
What is this needed for?

Reply:
@sotteson1 Can you explain your rationale behind this change?

Reply:
> @sotteson1 Can you explain your rationale behind this change?

It's been months since I added that, but it seems like there was a case where there was another string in the text after the first quotes were removed. I was returning false to indicate the function failed, but then it's probably not good that it's already modified the ref'ed parameter. Sorry I don't have more information than that.

Reply:
I see why this is needed.
We have a property ` Condition=" '$(VisualStudioVersion)' == '' or '$(VisualStudioVersion)' &lt; '14.0' "`
When we split this by `==` in the method `ConditionToDimensionValues`, the right side of the string becomes `'' or '$(VisualStudioVersion)' &lt; '14.0'`
The Unquote function ends up returning `' or '$(VisualStudioVersion)' &lt; '14.0` which is an invalid unquoted string

Reply:
A correct fix is checking whether the string s[1..^1] contains \` before we convert to unquoted string and not after.

Reply:
In an ideal scenario, the `ConditionToDimensionValues` method (on line 78 in the same file) should first split by the `or` and `and` conditions and then split by `==`
Do you think this might be a bug? @jmarolf what do you think is the expected behavior here?

Reply:
I think we should change the behavior in a followup PR to have `ConditionToDimensionValues` work as you describe.